modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryProcessor.java

```diff
@@ -1088,11 +1088,25 @@ else if (op0 instanceof SchemaAddQueryEntityOperation) {
     public void initQueryStructuresForNotStartedCache(DynamicCacheDescriptor cacheDesc) throws IgniteCheckedException {
         QuerySchema schema = cacheDesc.schema() != null ? cacheDesc.schema() : new QuerySchema();
 
+        if (ctx.discovery().localNode().isClient())
+            enrichConfiguration(cacheDesc);
```

Comment:
This place looks very suspicious, because cache configuration class has many fields marked with SerializeSeparately annotation. Here we deserialize all of them, but required only sqlFuncCls.

Reply:
I fixed this. The fix isn't ideal but it enables to pass a filter function to configuration enricher that allows only a specific fields to be deserialized.
Now only sqlFuncCls field will be deserialized by this call.